src/scan/postgres_scan.cpp

```diff
@@ -267,8 +284,13 @@ PostgresScanTableFunction::PostgresScanFunction(duckdb::ClientContext &, duckdb:
 
 		SlotGetAllAttrs(slot);
 		InsertTupleIntoChunk(output, local_state, slot);
+
+		if (MemoryContextMemAllocated(duckdb_pg_scan_memory_ctx, false) > 8 * 1024 * 1024L) {
+			MemoryContextReset(duckdb_pg_scan_memory_ctx);
+		}
```

Comment:
Is this one really needed? I'd rather keep it simple and only do the reset after we are done with the vector. Doing stuff with duckdb is always at the vector level. Then we also don't have to find a decent magic value, like the 8MB you use now.

Reply:
Yes, let's simplify by removing it. I was concerned about the wide table or large JSON columns. For example, a single JSON row can be several MB in size, and 2048 such rows would quickly consume a lot of memory. However, in this case, the DuckDB data chunk itself also consumes a significant amount of memory.